pkg/controller/worker/machines.go

```diff
@@ -206,3 +208,24 @@ func (w *workerDelegate) generateHashForWorkerPool(pool v1alpha1.WorkerPool) (st
 	// Generate the worker pool hash.
 	return worker.WorkerPoolHash(pool, w.cluster, nil, nil)
 }
+
+func (w *workerDelegate) getServerLabelsForMachine(machineType string, workerConfig *metalv1alpha1.WorkerConfig) (map[string]string, error) {
+	combinedLabels := make(map[string]string)
+	for _, t := range w.cloudProfileConfig.MachineTypes {
+		if t.Name == machineType {
+			for key, value := range t.ServerLabels {
+				combinedLabels[key] = value
+			}
+			break
+		}
+	}
+	if workerConfig.ExtraServerLabels != nil {
```

Comment:
Nitpick: nil doesn't need to be checked here, [iterating a nil map produces 0 iterations](https://go.dev/ref/spec#For_statements:~:text=If%20the%20map%20is%20nil%2C%20the%20number%20of%20iterations%20is%200.)

Reply:
fixed